nemo/collections/asr/parts/utils/diarization_utils.py

```diff
@@ -15,500 +15,194 @@
-        self.root_path = None
-        self.fix_word_ts_with_VAD = True
+        self.fix_word_ts_with_VAD = self.params['fix_word_ts_with_VAD']
+        self.root_path = cfg_diarizer['out_dir']
+        shutil.rmtree(self.root_path, ignore_errors=True)
+
+        self.vad_threshold_for_word_ts = 0.7
+        self.max_word_ts_length_in_sec = 0.6
+        self.cfg_diarizer = cfg_diarizer
+        self.word_ts_anchor_offset = 0.0
         self.run_ASR = None
+        self.ctm_exists = {}
         self.frame_VAD = {}
-        self.AUDIO_RTTM_MAP = {}
-
-    def set_asr_model(self, asr_model):
-        """
-        Setup the parameters for the given ASR model
-        Currently, the following models are supported:
-            stt_en_conformer_ctc_large
-            stt_en_conformer_ctc_medium
-            stt_en_conformer_ctc_small
-            QuartzNet15x5Base-En
-        """
-
-        if 'QuartzNet' in asr_model:
-            self.run_ASR = self.run_ASR_QuartzNet_CTC
-            asr_model = EncDecCTCModel.from_pretrained(model_name=asr_model, strict=False)
-            self.params['offset'] = -0.18
-            self.model_stride_in_secs = 0.02
-            self.asr_delay_sec = -1 * self.params['offset']
-
-        elif 'conformer_ctc' in asr_model:
-            self.run_ASR = self.run_ASR_BPE_CTC
-            asr_model = EncDecCTCModelBPE.from_pretrained(model_name=asr_model, strict=False)
-            self.model_stride_in_secs = 0.04
-            self.asr_delay_sec = 0.0
-            self.params['offset'] = 0
-            self.chunk_len_in_sec = 1.6
-            self.total_buffer_in_secs = 4
-
-        elif 'citrinet' in asr_model:
-            self.run_ASR = self.run_ASR_BPE_CTC
-            asr_model = EncDecCTCModelBPE.from_pretrained(model_name=asr_model, strict=False)
-            self.model_stride_in_secs = 0.08
-            self.asr_delay_sec = 0.0
-            self.params['offset'] = 0
-            self.chunk_len_in_sec = 1.6
-            self.total_buffer_in_secs = 4
-
-        elif 'conformer_transducer' in asr_model or 'contextnet' in asr_model:
-            self.run_ASR = self.run_ASR_BPE_RNNT
-            asr_model = EncDecRNNTBPEModel.from_pretrained(model_name=asr_model, strict=False)
-            self.model_stride_in_secs = 0.04
-            self.asr_delay_sec = 0.0
-            self.params['offset'] = 0
-            self.chunk_len_in_sec = 1.6
-            self.total_buffer_in_secs = 4
-        else:
-            raise ValueError(f"ASR model name not found: {asr_model}")
-        self.params['time_stride'] = self.model_stride_in_secs
-        self.asr_batch_size = 16
-        asr_model.eval()
-
+        self.align_error_list = []
+        self.AUDIO_RTTM_MAP = audio_rttm_map(self.manifest_filepath)
         self.audio_file_list = [value['audio_filepath'] for _, value in self.AUDIO_RTTM_MAP.items()]
+        if self.realigning_lm_params['arpa_language_model']:
+            self.realigning_lm = self.load_realigning_LM()
```

Comment:
why to load lm here even before diarization being performed?

Reply:
This part is moved to the actual realignment part.